spec/controllers/expenses_controller_spec.rb

```diff
@@ -0,0 +1,273 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ExpensesController, type: :request do
+  let!(:current_user) { create(:user) }
+
+  let(:account) { create(:account) }
+  let(:category) { create(:category) }
+
+  let!(:current_expense) { create(:expense, name: 'Expense', value: 10, user_id: current_user.id) }
+  let!(:second_expense) { create(:expense) }
+
+  let(:jwt_token) { JsonWebToken.encode(user_id: current_user.id) }
+  let(:json) { JSON.parse(response.body) }
+
+  describe '#index' do
+    subject { get(expenses_path, headers: { Authorization: jwt_token }) }
+
+    it 'returns all expenses from current user' do
+      subject
+
+      expect(response).to have_http_status :ok
+      expect(json.count).to eq(1)
+      expect(json[0]['name']).to include('Expense')
+      expect(json[0]['value_in_cents']).to eq(1000)
+      expect(Time.zone.parse(json[0]['paid_at'])).to eq(Time.parse('28-04-2022'))
+      expect(Time.zone.parse(json[0]['due_at'])).to eq(Time.parse('21-06-2022'))
+    end
+
+    include_examples 'Invalid JWT Token'
+  end
+
+  describe '#show' do
+    subject { get(expense_path(slug), headers: { Authorization: jwt_token }) }
+
+    let(:slug) { current_expense.slug }
+
+    context 'with valid params' do
+      it 'returns the given expense' do
+        subject
+
+        expect(response).to have_http_status :ok
+        expect(json['name']).to include('Expense')
+        expect(json['value_in_cents']).to eq(1000)
+      end
+    end
+
+    context 'with params for another user' do
+      let(:slug) { second_expense.slug }
+
+      it 'returns errors' do
+        subject
+
+        expect(response).to have_http_status :not_found
+        expect(json['errors']).to include('Expense not found')
+      end
+    end
+
+    context 'with invalid params' do
+      let(:slug) { 'invalid-slug' }
+
+      it 'returns errors' do
+        subject
+
+        expect(response).to have_http_status :not_found
+        expect(json['errors']).to include('Expense not found')
+      end
+    end
+
+    include_examples 'Invalid JWT Token'
+  end
+
+  describe '#create' do
+    subject { post(expenses_path, headers: { Authorization: jwt_token }, params: params) }
+
+    let(:name) { 'New expense name' }
+    let(:value) { 150 }
+    let(:due_at) { Time.parse('05-01-2023') }
+    let(:paid_at) { Time.parse('02-01-2023') }
```

Comment:
Se vc vai mandar uma POST request pra esse endpoint, isso aqui vai ter que ser uma string?

```suggestion
    let(:due_at) { '05-01-2023 16:20:00' }
    let(:paid_at) { '02-01-2023 12:00:00' }
```

Reply:
Acredito que sim... sendo tudo no timezone `UTC` 🤔